sdk/dslink/src/main/java/org/dsa/iot/dslink/DSLinkHandler.java

```diff
@@ -91,6 +92,43 @@ public Configuration getConfig() {
     }
 
     /**
+     * Nodes in links can specify an icon with the @icon attribute.  The default
+     * implementation of this method looks for the attribute value in the icons
+     * directory in the root of the jar containing the DSLinkHandler subclass
+     * getClass().getResourceAsStream("/icons/" + attributeValue).
+     *
+     * @param name Name or path of the desired icon.
+     * @return The bytes representing the icon image, or null if there are any issues.
+     */
+    public byte[] getIcon(String name) {
+        StringBuilder builder = new StringBuilder("/icons");
+        if (name.charAt(0) != '/') {
+            builder.append('/');
+        }
+        builder.append(name);
+        InputStream in = getClass().getResourceAsStream(builder.toString());
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try {
+            byte[] buf = new byte[512];
+            int len = in.read(buf);
+            while (len > 0) {
+                out.write(buf, 0, len);
+                len = in.read(buf);
+            }
+        } catch (Exception x) {
+            //The lack of an icon will be enough notification
```

Comment:
Not sure I like this. I'd rather failfast, than assume the icon is there, when it's not.